<a href="https://colab.research.google.com/github/jonafel/Tugas-Sentiment-Analysis/blob/main/tugas_akhir_sentimen.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

##TUGAS AKHIR SENTIMEN ANALISIS JONATHAN FELIX

Melalui Collabs ini saya mencoba menganalisa sentimen komentar yang ada di akun Instagram [Indozone.ID](https://www.instagram.com/p/DEmxFKTzfqm/)

Tapi sebelum cek langsung ke komentar dari Indozone, ayo kita train data sementara cyberbullying. Kita juga harus menyiapkan semua Library yang akan dibutuhkan

In [54]:
!pip install Sastrawi

   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 209.7/209.7 kB 3.6 MB/s eta 0:00:00


In [2]:
!pip install nltk

In [55]:
import pandas as pd
import nltk
import re
import string
from collections import Counter
from Sastrawi.Stemmer.StemmerFactory import StemmerFactory

nltk.download('popular')
nltk.download('punkt')
nltk.download('punkt_tab')

[nltk_data] Downloading collection 'popular'
[nltk_data]    | 
[nltk_data]    | Downloading package cmudict to /root/nltk_data...
[nltk_data]    |   Package cmudict is already up-to-date!
[nltk_data]    | Downloading package gazetteers to /root/nltk_data...
[nltk_data]    |   Package gazetteers is already up-to-date!
[nltk_data]    | Downloading package genesis to /root/nltk_data...
[nltk_data]    |   Package genesis is already up-to-date!
[nltk_data]    | Downloading package gutenberg to /root/nltk_data...
[nltk_data]    |   Package gutenberg is already up-to-date!
[nltk_data]    | Downloading package inaugural to /root/nltk_data...
[nltk_data]    |   Package inaugural is already up-to-date!
[nltk_data]    | Downloading package movie_reviews to
[nltk_data]    |     /root/nltk_data...
[nltk_data]    |   Package movie_reviews is already up-to-date!
[nltk_data]    | Downloading package names to /root/nltk_data...
[nltk_data]    |   Package names is already up-to-date!
[nltk_data]    | Do

True

## IMPORT DATA TRAINING & TEXT CLEANING

[dataset CyberBullying](https://drive.google.com/file/d/10mhzsF0lWdc4rqqowsb50K9ZZ_nOwzA8/view?usp=drive_link)

In [106]:
data = pd.read_csv('komentar_instagram_cyberbullying.csv')
data

,Id,Sentiment,Instagram Comment Text
0,1,negative,<USERNAME> TOLOL!! Gak ada hubungan nya kegug...
1,2,negative,Geblek lo tata...cowo bgt dibela2in balikan......
2,3,negative,Kmrn termewek2 skr lengket lg duhhh kok labil ...
3,4,negative,"Intinya kalau kesel dengan ATT nya, gausah ke ..."
4,5,negative,"hadewwwww permpuan itu lg!!!!sakit jiwa,knp ha..."
...,...,...,...
395,396,positive,Bangga sama suami yg selalu ingat istri disela...
396,397,positive,Apaoun pekerjaannya yg penting halal u tuk men...
397,398,positive,Gojek itu mayoritas pegangguran yang lama gak ...
398,399,positive,<USERNAME> aslinya cantik dan ayu loh mbak kr...


In [5]:
def text_cleaning(text:str):
  text_cleaned = text.lower()

  pattern = re.compile(r'@\w+')
  clean_url_text = pattern.sub("", text_cleaned)

  clean_number_text = re.sub(r"[\d]+.[\d]+|[\d]+","", clean_url_text)

  pattern = re.compile(r'\(?https?://[^\s$#]+\)?')
  clean_url_text = pattern.sub("", clean_number_text)


  clean_punct_text = clean_url_text.translate(str.maketrans('','',string.punctuation))

  clean_wspace_text = clean_punct_text.strip()
  clean_wspace_text = clean_wspace_text.replace('\n','')
  clean_wspace_text = re.sub(r"[\s][.,]?[\s]"," ",clean_wspace_text)

  clean_wspace_text = re.sub(r'\b(\w+)\1\b', r'\1', clean_wspace_text)

  return clean_wspace_text

[StopWord Bahasa Indonesia](https://drive.google.com/file/d/1E844kUnMSCguYSQcNZzj62aJjQX1HwVO/view?usp=drive_link)

In [6]:
def load_stopwords(filepath):
      try:
          df_stopword = pd.read_csv('stopwordbahasa.csv', encoding='utf-8')
          stopword_list = df_stopword.iloc[:,0].tolist()
          stopword_set = set(stopword_list)

          return stopword_set
      except FileNotFoundError:
          print(f"File not found: {filepath}")
          return set()

In [7]:
def stopword_cleaning(text, stopword_set):
        text_words = text.split()
        cleaned_words = [word for word in text_words if word not in stopword_set]
        return ' '.join(cleaned_words)

In [8]:
stopword_sets = load_stopwords('stopwordbahasa.csv')
stopword_sets

{'adalah',
 'adanya',
 'adapun',
 'agak',
 'agaknya',
 'agar',
 'akan',
 'akankah',
 'akhir',
 'akhiri',
 'akhirnya',
 'aku',
 'akulah',
 'amat',
 'amatlah',
 'anda',
 'andalah',
 'antar',
 'antara',
 'antaranya',
 'apa',
 'apaan',
 'apabila',
 'apakah',
 'apalagi',
 'apatah',
 'artinya',
 'asal',
 'asalkan',
 'atas',
 'atau',
 'ataukah',
 'ataupun',
 'awal',
 'awalnya',
 'bagai',
 'bagaikan',
 'bagaimana',
 'bagaimanakah',
 'bagaimanapun',
 'bagi',
 'bagian',
 'bahkan',
 'bahwa',
 'bahwasanya',
 'baik',
 'bakal',
 'bakalan',
 'balik',
 'banyak',
 'bapak',
 'baru',
 'bawah',
 'beberapa',
 'begini',
 'beginian',
 'beginikah',
 'beginilah',
 'begitu',
 'begitukah',
 'begitulah',
 'begitupun',
 'bekerja',
 'belakang',
 'belakangan',
 'belum',
 'belumlah',
 'benar',
 'benarkah',
 'benarlah',
 'berada',
 'berakhir',
 'berakhirlah',
 'berakhirnya',
 'berapa',
 'berapakah',
 'berapalah',
 'berapapun',
 'berarti',
 'berawal',
 'berbagai',
 'berdatangan',
 'beri',
 'berikan',
 'berikut',
 'beri

Kita bersihkan data dari Kolom Instagram Comment Text

In [107]:
data['text_cleaned'] = data['Instagram Comment Text'].apply(text_cleaning)
data

,Id,Sentiment,Instagram Comment Text,text_cleaned
0,1,negative,<USERNAME> TOLOL!! Gak ada hubungan nya kegug...,username tolol gak ada hubungan nya keguguran ...
1,2,negative,Geblek lo tata...cowo bgt dibela2in balikan......,geblek lo tatacowo bgt dibelain balikanhadewwn...
2,3,negative,Kmrn termewek2 skr lengket lg duhhh kok labil ...,kmrn termewek skr lengket lg duhhh kok labil b...
3,4,negative,"Intinya kalau kesel dengan ATT nya, gausah ke ...",intinya kalau kesel dengan att nya gausah ke a...
4,5,negative,"hadewwwww permpuan itu lg!!!!sakit jiwa,knp ha...",hadewwwww permpuan itu lgsakit jiwaknp harus d...
...,...,...,...,...
395,396,positive,Bangga sama suami yg selalu ingat istri disela...,bangga sama suami yg selalu ingat istri disela...
396,397,positive,Apaoun pekerjaannya yg penting halal u tuk men...,apaoun pekerjaannya yg penting halal u tuk men...
397,398,positive,Gojek itu mayoritas pegangguran yang lama gak ...,gojek itu mayoritas pegangguran yang lama gak ...
398,399,positive,<USERNAME> aslinya cantik dan ayu loh mbak kr...,username aslinya cantik dan ayu loh mbak krn a...


In [108]:
data['text_non_stopword'] = data['text_cleaned'].apply(stopword_cleaning, stopword_set=stopword_sets)
data

,Id,Sentiment,Instagram Comment Text,text_cleaned,text_non_stopword
0,1,negative,<USERNAME> TOLOL!! Gak ada hubungan nya kegug...,username tolol gak ada hubungan nya keguguran ...,username tolol gak ada hubungan nya keguguran ...
1,2,negative,Geblek lo tata...cowo bgt dibela2in balikan......,geblek lo tatacowo bgt dibelain balikanhadewwn...,geblek lo tatacowo bgt dibelain balikanhadewwn...
2,3,negative,Kmrn termewek2 skr lengket lg duhhh kok labil ...,kmrn termewek skr lengket lg duhhh kok labil b...,kmrn termewek skr lengket lg duhhh labil bgt s...
3,4,negative,"Intinya kalau kesel dengan ATT nya, gausah ke ...",intinya kalau kesel dengan att nya gausah ke a...,intinya kesel att nya gausah anaknya kasian pe...
4,5,negative,"hadewwwww permpuan itu lg!!!!sakit jiwa,knp ha...",hadewwwww permpuan itu lgsakit jiwaknp harus d...,hadewwwww permpuan lgsakit jiwaknp yg jd peran...
...,...,...,...,...,...
395,396,positive,Bangga sama suami yg selalu ingat istri disela...,bangga sama suami yg selalu ingat istri disela...,bangga suami yg istri disela apapun profesimu ...
396,397,positive,Apaoun pekerjaannya yg penting halal u tuk men...,apaoun pekerjaannya yg penting halal u tuk men...,apaoun pekerjaannya yg halal u tuk menafkahi k...
397,398,positive,Gojek itu mayoritas pegangguran yang lama gak ...,gojek itu mayoritas pegangguran yang lama gak ...,gojek mayoritas pegangguran gak kerja trus put...
398,399,positive,<USERNAME> aslinya cantik dan ayu loh mbak kr...,username aslinya cantik dan ayu loh mbak krn a...,username aslinya cantik ayu loh mbak krn berte...


Setelah dibersihkan dari tanda baca, huruf besar kecilnya dan Stopword, kita cek kata yang peling sering muncul

In [34]:
from nltk.tokenize import word_tokenize
tokens = data['text_non_stopword'].tolist()

In [27]:
list_of_list = [word_tokenize(sentence) for sentence in tokens]
list_of_token = [word for sentence in list_of_list for word in sentence]

print(list_of_list)
print(list_of_token)

[['username', 'tolol', 'gak', 'ada', 'hubungan', 'nya', 'keguguran', 'dgn', 'pake', 'hijab', 'syari', 'yg', 'lo', 'bilang', 'bayi', 'nya', 'kepanasan', 'didalem', 'gak', 'ada', 'hubungan', 'nya', 'woyyyy', 'otak', 'jempol', 'lo', 'gak', 'singkron', 'sih', 'ya', 'nulis', 'komentar'], ['geblek', 'lo', 'tatacowo', 'bgt', 'dibelain', 'balikanhadewwntar', 'ditinggal', 'lg', 'nyalahin', 'tuh', 'cowopadahal', 'kitenya', 'yg', 'oon'], ['kmrn', 'termewek', 'skr', 'lengket', 'lg', 'duhhh', 'labil', 'bgt', 'sih', 'mbak', 'kya', 'abege', 'ajah', 'ato', 'yg', 'kmrn', 'cari', 'sensasi', 'biar', 'top', 'markotoppp', 'ertong', 'gk', 'berkualitas'], ['intinya', 'kesel', 'att', 'nya', 'gausah', 'anaknya', 'kasian', 'perkembangan', 'psikis', 'anak', 'kedepannya', 'orang', 'bener', 'bener', 'tolol', 'skrg', 'seandainya', 'anak', 'anaknya', 'dikatain', 'orang', 'benci', 'gimana', 'perasaan', 'benci', 'tau', 'batesnya', 'namanya', 'manusia', 'gaakan', 'penyuka', 'ada', 'haters'], ['hadewwwww', 'permpuan', '

In [28]:
word_counts = Counter(list_of_token)

for word, count in word_counts.most_common(100):
  print(f"{word}: {count}")

yg: 211
username: 86
nya: 85
ya: 85
ga: 77
aja: 67
gak: 63
ada: 55
anak: 54
orang: 53
banget: 47
si: 44
artis: 42
org: 41
semoga: 40
cantik: 39
bgt: 34
jd: 34
kaya: 33
udah: 31
kalo: 29
suka: 29
mba: 29
klo: 28
sm: 27
suami: 26
lo: 25
sih: 24
laki: 24
lu: 24
allah: 24
mbak: 23
tau: 23
jg: 23
biar: 22
udh: 22
pelakor: 22
istri: 22
keluarga: 22
n: 22
pake: 21
deh: 21
bagus: 21
tp: 21
nikah: 20
sok: 20
ama: 20
manusia: 19
gue: 19
ganteng: 19
bilang: 18
sehat: 18
skrg: 17
becus: 17
ayu: 17
perempuan: 17
lg: 16
anaknya: 16
suaranya: 16
dunia: 16
tuh: 15
kali: 15
dr: 15
gitu: 15
dasar: 15
mah: 15
keren: 15
otak: 14
bener: 14
d: 14
bikin: 14
emang: 14
bego: 13
ni: 13
bahagia: 13
aurel: 13
att: 12
jijik: 12
mukanya: 12
liat: 12
kampungan: 12
gw: 12
jevin: 12
kek: 11
jaman: 11
malu: 11
ma: 11
yaa: 11
ny: 11
muda: 11
kayak: 11
jgn: 11
y: 11
tu: 11
gk: 10
krn: 10
nih: 10
gua: 10
aj: 10
g: 10


In [29]:
from nltk.util import ngrams

bigram = ngrams(list_of_token,2)
bigram_count = Counter(bigram)

for bigram, count in bigram_count.most_common(20):
  print(f"{bigram[0]} {bigram[1]} = {count}")

gak ada = 8
ada yg = 8
jaman now = 6
biar ga = 6
ga becus = 6
becus perempuan = 6
anak nya = 6
org yg = 6
allah swt = 6
dunia akhirat = 5
bgt sm = 5
perempuan yg = 5
laki orang = 5
yg suka = 5
ga ada = 5
artis yg = 5
yg yg = 4
kids jaman = 4
nya yg = 4
sok suci = 4


Dari cek kata yang sering keluar dan pasangan kata yang sering keluar, kita bisa rapihkan kata" yang merupakan slang atau singkatan dari bahasa Indonesia. da dibawah ini ada banyak kata yang kita rapihkan.

In [38]:
word_dict = {
    "ga":"tidak",
    "gak":"tidak",
    "kalo":"jika",
    "pelakor": "perebut laki orang",
    "bgt" : "sangat",
    "dgn": "dengan",
    "tp":"tapi",
    "sok":"sombong",
    "dr":"dari",
    "gw":"saya",
    'gua':'saya',
    'gue':'saya',
    "bener":"benar",
    "sy":"saya",
    'lo':'kamu',
    'lu':'kamu',
    'org':'orang',
    'n':'dan',
    'pake':'pakai',
    'jd':'jadi',
    'laki':'pria',
    'sm':'sama',
    'yg':'yang',
    'tau':'tahu',
    'd':'di',
    'tolol':'bodoh',
    'bego':'bodoh',
    'att':'atitut',
    'jgn':'jangan',
    'y':'ya',
    'tu':'tuh',
    'krn':'karena',
    'jaman':'zaman',
    'aj':'saja',
    'g':'tidak',
    'now':'sekarang',
    'biar':'agar',
    'geblek':'bodoh',
    'kmrn':'kemarin',
    'skrg':'sekarang',
    'apaoun':'apapun',
    'trus':'lalu'
}

def fix_slang(sentence, word_dict):
  tokens = word_tokenize(sentence)
  fix_slangs = [word_dict.get(word, word) for word in tokens]
  return ' '.join(fix_slangs)


In [109]:
data['text_non_slang']  = data['text_non_stopword'] .apply(lambda x: fix_slang(x, word_dict))
data

,Id,Sentiment,Instagram Comment Text,text_cleaned,text_non_stopword,text_non_slang
0,1,negative,<USERNAME> TOLOL!! Gak ada hubungan nya kegug...,username tolol gak ada hubungan nya keguguran ...,username tolol gak ada hubungan nya keguguran ...,username bodoh tidak ada hubungan nya kegugura...
1,2,negative,Geblek lo tata...cowo bgt dibela2in balikan......,geblek lo tatacowo bgt dibelain balikanhadewwn...,geblek lo tatacowo bgt dibelain balikanhadewwn...,bodoh kamu tatacowo sangat dibelain balikanhad...
2,3,negative,Kmrn termewek2 skr lengket lg duhhh kok labil ...,kmrn termewek skr lengket lg duhhh kok labil b...,kmrn termewek skr lengket lg duhhh labil bgt s...,kemarin termewek skr lengket lg duhhh labil sa...
3,4,negative,"Intinya kalau kesel dengan ATT nya, gausah ke ...",intinya kalau kesel dengan att nya gausah ke a...,intinya kesel att nya gausah anaknya kasian pe...,intinya kesel atitut nya gausah anaknya kasian...
4,5,negative,"hadewwwww permpuan itu lg!!!!sakit jiwa,knp ha...",hadewwwww permpuan itu lgsakit jiwaknp harus d...,hadewwwww permpuan lgsakit jiwaknp yg jd peran...,hadewwwww permpuan lgsakit jiwaknp yang jadi p...
...,...,...,...,...,...,...
395,396,positive,Bangga sama suami yg selalu ingat istri disela...,bangga sama suami yg selalu ingat istri disela...,bangga suami yg istri disela apapun profesimu ...,bangga suami yang istri disela apapun profesim...
396,397,positive,Apaoun pekerjaannya yg penting halal u tuk men...,apaoun pekerjaannya yg penting halal u tuk men...,apaoun pekerjaannya yg halal u tuk menafkahi k...,apapun pekerjaannya yang halal u tuk menafkahi...
397,398,positive,Gojek itu mayoritas pegangguran yang lama gak ...,gojek itu mayoritas pegangguran yang lama gak ...,gojek mayoritas pegangguran gak kerja trus put...,gojek mayoritas pegangguran tidak kerja lalu p...
398,399,positive,<USERNAME> aslinya cantik dan ayu loh mbak kr...,username aslinya cantik dan ayu loh mbak krn a...,username aslinya cantik ayu loh mbak krn berte...,username aslinya cantik ayu loh mbak karena be...


Setelah dapat hasil bersihnya bari bisa kita lanjutkan ke 2 Metode yang akan kita bandingkan

## METODE NAIVE BAYES

[Kata Positif](https://drive.google.com/file/d/1YW3fqnBDeBa6-UF5HjZWCjqbENOwdSqp/view?usp=drive_link)
[Kata Negatif](https://drive.google.com/file/d/1mJkAtp0rWRSZnpvH0JwC27fZCKvF_UXP/view?usp=drive_link)

In [40]:
with open("./kata positif.txt","r") as positif_file :
  positive_words = list(set(positif_file.readlines()))

with open("./kata negatif.txt","r") as negative_file :
  negative_words = list(set(negative_file.readlines()))

In [41]:
positive_words = [word.rstrip() for word in positive_words]
negative_words = [word.rstrip() for word in negative_words]

print(positive_words[:10])
print(negative_words[:10])

['dipathui', 'sportivitas', 'pencarian', 'dikokohkan', 'dibekali', 'hormat', 'tersukses', 'loyal', 'tuntas', 'mempersukses']
['tsunami', 'ugal-ugalan', 'traumatis', 'cidera', 'goyah', 'menangisi', 'kecelakaan', 'dimandulkan', 'mengorbankan', 'asal bunyi']


In [48]:
def sentiment_counts(text, positive_words, negative_words):
  good_count = 0
  bad_count = 0

  tokens = word_tokenize(text)

  for token in tokens:
        if token in positive_words:
            good_count += 1
        if token in negative_words:
            bad_count += 1

  return good_count - bad_count

In [49]:
data['count'] = data['text_non_slang'].apply(lambda x: sentiment_counts(x, positive_words, negative_words))

In [50]:
data

,Id,Sentiment,Instagram Comment Text,text_cleaned,text_non_stopword,text_non_slang,count
0,1,negative,<USERNAME> TOLOL!! Gak ada hubungan nya kegug...,username tolol gak ada hubungan nya keguguran ...,username tolol gak ada hubungan nya keguguran ...,username bodoh tidak ada hubungan nya kegugura...,-3
1,2,negative,Geblek lo tata...cowo bgt dibela2in balikan......,geblek lo tatacowo bgt dibelain balikanhadewwn...,geblek lo tatacowo bgt dibelain balikanhadewwn...,bodoh kamu tatacowo sangat dibelain balikanhad...,-1
2,3,negative,Kmrn termewek2 skr lengket lg duhhh kok labil ...,kmrn termewek skr lengket lg duhhh kok labil b...,kmrn termewek skr lengket lg duhhh labil bgt s...,kemarin termewek skr lengket lg duhhh labil sa...,0
3,4,negative,"Intinya kalau kesel dengan ATT nya, gausah ke ...",intinya kalau kesel dengan att nya gausah ke a...,intinya kesel att nya gausah anaknya kasian pe...,intinya kesel atitut nya gausah anaknya kasian...,1
4,5,negative,"hadewwwww permpuan itu lg!!!!sakit jiwa,knp ha...",hadewwwww permpuan itu lgsakit jiwaknp harus d...,hadewwwww permpuan lgsakit jiwaknp yg jd peran...,hadewwwww permpuan lgsakit jiwaknp yang jadi p...,-1
...,...,...,...,...,...,...,...
395,396,positive,Bangga sama suami yg selalu ingat istri disela...,bangga sama suami yg selalu ingat istri disela...,bangga suami yg istri disela apapun profesimu ...,bangga suami yang istri disela apapun profesim...,3
396,397,positive,Apaoun pekerjaannya yg penting halal u tuk men...,apaoun pekerjaannya yg penting halal u tuk men...,apaoun pekerjaannya yg halal u tuk menafkahi k...,apapun pekerjaannya yang halal u tuk menafkahi...,2
397,398,positive,Gojek itu mayoritas pegangguran yang lama gak ...,gojek itu mayoritas pegangguran yang lama gak ...,gojek mayoritas pegangguran gak kerja trus put...,gojek mayoritas pegangguran tidak kerja lalu p...,3
398,399,positive,<USERNAME> aslinya cantik dan ayu loh mbak kr...,username aslinya cantik dan ayu loh mbak krn a...,username aslinya cantik ayu loh mbak krn berte...,username aslinya cantik ayu loh mbak karena be...,1


pada kolom count, bisa kita lihat banyak angka yang menunjukkan kearah mana kalimat di depannya positif atau negatif. jadi kita buat kolom sendiri yang dapat memberikan hasil sentimennya.

In [51]:
def sentimen(score):
  if score > 0:
    return "positif"
  elif score == 0:
    return "netral"
  else:
    return "negatif"

In [52]:
data['hasil sentimen'] = data['count'].apply(sentimen)
data

,Id,Sentiment,Instagram Comment Text,text_cleaned,text_non_stopword,text_non_slang,count,hasil sentimen
0,1,negative,<USERNAME> TOLOL!! Gak ada hubungan nya kegug...,username tolol gak ada hubungan nya keguguran ...,username tolol gak ada hubungan nya keguguran ...,username bodoh tidak ada hubungan nya kegugura...,-3,negatif
1,2,negative,Geblek lo tata...cowo bgt dibela2in balikan......,geblek lo tatacowo bgt dibelain balikanhadewwn...,geblek lo tatacowo bgt dibelain balikanhadewwn...,bodoh kamu tatacowo sangat dibelain balikanhad...,-1,negatif
2,3,negative,Kmrn termewek2 skr lengket lg duhhh kok labil ...,kmrn termewek skr lengket lg duhhh kok labil b...,kmrn termewek skr lengket lg duhhh labil bgt s...,kemarin termewek skr lengket lg duhhh labil sa...,0,netral
3,4,negative,"Intinya kalau kesel dengan ATT nya, gausah ke ...",intinya kalau kesel dengan att nya gausah ke a...,intinya kesel att nya gausah anaknya kasian pe...,intinya kesel atitut nya gausah anaknya kasian...,1,positif
4,5,negative,"hadewwwww permpuan itu lg!!!!sakit jiwa,knp ha...",hadewwwww permpuan itu lgsakit jiwaknp harus d...,hadewwwww permpuan lgsakit jiwaknp yg jd peran...,hadewwwww permpuan lgsakit jiwaknp yang jadi p...,-1,negatif
...,...,...,...,...,...,...,...,...
395,396,positive,Bangga sama suami yg selalu ingat istri disela...,bangga sama suami yg selalu ingat istri disela...,bangga suami yg istri disela apapun profesimu ...,bangga suami yang istri disela apapun profesim...,3,positif
396,397,positive,Apaoun pekerjaannya yg penting halal u tuk men...,apaoun pekerjaannya yg penting halal u tuk men...,apaoun pekerjaannya yg halal u tuk menafkahi k...,apapun pekerjaannya yang halal u tuk menafkahi...,2,positif
397,398,positive,Gojek itu mayoritas pegangguran yang lama gak ...,gojek itu mayoritas pegangguran yang lama gak ...,gojek mayoritas pegangguran gak kerja trus put...,gojek mayoritas pegangguran tidak kerja lalu p...,3,positif
398,399,positive,<USERNAME> aslinya cantik dan ayu loh mbak kr...,username aslinya cantik dan ayu loh mbak krn a...,username aslinya cantik ayu loh mbak krn berte...,username aslinya cantik ayu loh mbak karena be...,1,positif


In [53]:
data['hasil sentimen'].value_counts()

,count
hasil sentimen,
positif,172
negatif,123
netral,105


Melalui Metode Naive Bayes, Hasil Positif dari dataset Training yang saya gunakan ada 172. Negatif di 123 dan Netral di 105.

##METODE LEXICON

Kita coba data yang sama tapi dengam metode yang berbeda
[Positif Lexicon](https://drive.google.com/file/d/1JnwJW4aQCJTZ5N2gtSOb-Dmcu1Pnamt8/view?usp=drive_link)
[Negatif Lexicon](https://drive.google.com/file/d/1gQRv8cFpBDNSPy5MYUzo3d-Z4Aaswl_M/view?usp=drive_link)

In [57]:
factory = StemmerFactory()
stemmer = factory.create_stemmer()

with open("./lex positif.txt","r") as positif_file :
  positive_words_lex = list(set(positif_file.readlines()))

with open("./lex negatif.txt","r") as negative_file :
  negative_words_lex = list(set(negative_file.readlines()))

positive_lex = [word.rstrip() for word in positive_words_lex]
negative_lex = [word.rstrip() for word in negative_words_lex]


In [58]:
def sentiment_counts_lex(text, positive_lex, negative_lex, stemmer):

  tokens = word_tokenize(text)
  stemmed_tokens = [stemmer.stem(token) for token in tokens]

  positive_count = 0
  negative_count = 0

  for token in stemmed_tokens:
        if token in positive_lex:
            positive_count += 1
        if token in negative_lex:
            negative_count += 1

  sentiment_score = positive_count - negative_count

  if sentiment_score > 0:
        return "positive"
  elif sentiment_score == 0:
        return "neutral"
  else:
        return "negative"

In [61]:
data['lexicon_sentiment'] = data['text_non_slang'].apply(lambda x: sentiment_counts_lex(x, positive_lex, negative_lex, stemmer))
data

,Id,Sentiment,Instagram Comment Text,text_cleaned,text_non_stopword,text_non_slang,count,hasil sentimen,lexicon_sentiment
0,1,negative,<USERNAME> TOLOL!! Gak ada hubungan nya kegug...,username tolol gak ada hubungan nya keguguran ...,username tolol gak ada hubungan nya keguguran ...,username bodoh tidak ada hubungan nya kegugura...,-3,negatif,negative
1,2,negative,Geblek lo tata...cowo bgt dibela2in balikan......,geblek lo tatacowo bgt dibelain balikanhadewwn...,geblek lo tatacowo bgt dibelain balikanhadewwn...,bodoh kamu tatacowo sangat dibelain balikanhad...,-1,negatif,negative
2,3,negative,Kmrn termewek2 skr lengket lg duhhh kok labil ...,kmrn termewek skr lengket lg duhhh kok labil b...,kmrn termewek skr lengket lg duhhh labil bgt s...,kemarin termewek skr lengket lg duhhh labil sa...,0,netral,negative
3,4,negative,"Intinya kalau kesel dengan ATT nya, gausah ke ...",intinya kalau kesel dengan att nya gausah ke a...,intinya kesel att nya gausah anaknya kasian pe...,intinya kesel atitut nya gausah anaknya kasian...,1,positif,negative
4,5,negative,"hadewwwww permpuan itu lg!!!!sakit jiwa,knp ha...",hadewwwww permpuan itu lgsakit jiwaknp harus d...,hadewwwww permpuan lgsakit jiwaknp yg jd peran...,hadewwwww permpuan lgsakit jiwaknp yang jadi p...,-1,negatif,neutral
...,...,...,...,...,...,...,...,...,...
395,396,positive,Bangga sama suami yg selalu ingat istri disela...,bangga sama suami yg selalu ingat istri disela...,bangga suami yg istri disela apapun profesimu ...,bangga suami yang istri disela apapun profesim...,3,positif,positive
396,397,positive,Apaoun pekerjaannya yg penting halal u tuk men...,apaoun pekerjaannya yg penting halal u tuk men...,apaoun pekerjaannya yg halal u tuk menafkahi k...,apapun pekerjaannya yang halal u tuk menafkahi...,2,positif,positive
397,398,positive,Gojek itu mayoritas pegangguran yang lama gak ...,gojek itu mayoritas pegangguran yang lama gak ...,gojek mayoritas pegangguran gak kerja trus put...,gojek mayoritas pegangguran tidak kerja lalu p...,3,positif,neutral
398,399,positive,<USERNAME> aslinya cantik dan ayu loh mbak kr...,username aslinya cantik dan ayu loh mbak krn a...,username aslinya cantik ayu loh mbak krn berte...,username aslinya cantik ayu loh mbak karena be...,1,positif,positive


In [62]:
data['lexicon_sentiment'].value_counts()

,count
lexicon_sentiment,
positive,154
negative,151
neutral,95


Proses yang sama tapi hasil yang berbeda walaupun tetap komentar sentimen positif ada di nomor 1, diikuti oleh negatif dan netral. Hasil disini jauh lebih kompetitif

## Sentiment Analysis pada komentar di postingan IG [Indozone ID](https://www.instagram.com/p/DEmxFKTzfqm/)

Ayo kita coba dengan 50 komentar acak dari postingan di IG Indozone diatas [Komentar Indozone](https://drive.google.com/file/d/1hTv6Ts6gVEsJwRUn3jVFh4jpNlDgRwmv/view?usp=drive_link)

In [82]:
with open('Komentar Indozone txt.txt', 'r') as file:
    komentar_indozone = file.readlines()

df = pd.DataFrame(komentar_indozone, columns=['Instagram Comment Text'])
print(df)

                               Instagram Comment Text
0                           Instagram Comment Text\n
1   Setiap mau Ramadhan sama IdulFitri ada aja pro...
2               Penyakit mematikan itu adalah Media\n
3                         Hahahah berita jancok ini\n
4   Apaan si media bikin ribet bae di cina biasa2 ...
5   Kata teman gua yg tinggal dichina dsna B aja c...
6   Heran aku heran tadi pas natal dan tahun baru ...
7   Media2 memperkeruh kalo emg di anggap biasa aj...
8                    Media emang suka nakut-nakutin\n
9   Yg nama a perubahan cuaca uda pasti sakit apa ...
10               di indo langsung jadi lahan bisnis\n
11  Kami di malaysia ga ada apa2 ga ada dngr kata2...
12  BODO AMAT !!! Setiap mau puasa setiap mau leba...
13  Beritanya jangan yang bikin takut saya lagi ca...
14  Byk di yutube org org indo yg di cina bilang g...
15  Apaan banyak video orang indo di China pada bi...
16  ya elaah, yakali gue sakit tenggorokan gara2 b...
17  Bisnis yang menguntungka

Ternyata Kolom Pertamanya masih ikut dan mari kita drop

In [83]:
df= df.drop(index=0)
df

,Instagram Comment Text
1,Setiap mau Ramadhan sama IdulFitri ada aja pro...
2,Penyakit mematikan itu adalah Media\n
3,Hahahah berita jancok ini\n
4,Apaan si media bikin ribet bae di cina biasa2 ...
5,Kata teman gua yg tinggal dichina dsna B aja c...
6,Heran aku heran tadi pas natal dan tahun baru ...
7,Media2 memperkeruh kalo emg di anggap biasa aj...
8,Media emang suka nakut-nakutin\n
9,Yg nama a perubahan cuaca uda pasti sakit apa ...
10,di indo langsung jadi lahan bisnis\n


## Pembersihan Data Komentar

dengan metode yang sama seperti data training kita jalankan pembersihannya

In [84]:
df['text_cleaned'] = df['Instagram Comment Text'].apply(text_cleaning)
df

,Instagram Comment Text,text_cleaned
1,Setiap mau Ramadhan sama IdulFitri ada aja pro...,setiap mau ramadhan sama idulfitri ada aja pro...
2,Penyakit mematikan itu adalah Media\n,penyakit mematikan itu adalah media
3,Hahahah berita jancok ini\n,hahahah berita jancok ini
4,Apaan si media bikin ribet bae di cina biasa2 ...,apaan si media bikin ribet bae di cina biasa a...
5,Kata teman gua yg tinggal dichina dsna B aja c...,kata teman gua yg tinggal dichina dsna b aja c...
6,Heran aku heran tadi pas natal dan tahun baru ...,heran aku heran tadi pas natal dan tahun baru ...
7,Media2 memperkeruh kalo emg di anggap biasa aj...,media memperkeruh kalo emg di anggap biasa aja...
8,Media emang suka nakut-nakutin\n,media emang suka nakutnakutin
9,Yg nama a perubahan cuaca uda pasti sakit apa ...,yg nama a perubahan cuaca uda pasti sakit apa ...
10,di indo langsung jadi lahan bisnis\n,di indo langsung jadi lahan bisnis


In [85]:
df['text_non_stopword'] = df['text_cleaned'].apply(stopword_cleaning, stopword_set=stopword_sets)
df

,Instagram Comment Text,text_cleaned,text_non_stopword
1,Setiap mau Ramadhan sama IdulFitri ada aja pro...,setiap mau ramadhan sama idulfitri ada aja pro...,ramadhan idulfitri ada aja program virus ya
2,Penyakit mematikan itu adalah Media\n,penyakit mematikan itu adalah media,penyakit mematikan media
3,Hahahah berita jancok ini\n,hahahah berita jancok ini,hahahah berita jancok
4,Apaan si media bikin ribet bae di cina biasa2 ...,apaan si media bikin ribet bae di cina biasa a...,si media bikin ribet bae cina aja jg ga bikin ...
5,Kata teman gua yg tinggal dichina dsna B aja c...,kata teman gua yg tinggal dichina dsna b aja c...,teman gua yg tinggal dichina dsna b aja indo a...
6,Heran aku heran tadi pas natal dan tahun baru ...,heran aku heran tadi pas natal dan tahun baru ...,heran heran pas natal kemana aja giliran ramad...
7,Media2 memperkeruh kalo emg di anggap biasa aj...,media memperkeruh kalo emg di anggap biasa aja...,media memperkeruh kalo emg anggap aja udah sto...
8,Media emang suka nakut-nakutin\n,media emang suka nakutnakutin,media emang suka nakutnakutin
9,Yg nama a perubahan cuaca uda pasti sakit apa ...,yg nama a perubahan cuaca uda pasti sakit apa ...,yg nama a perubahan cuaca uda sakit imun lemah...
10,di indo langsung jadi lahan bisnis\n,di indo langsung jadi lahan bisnis,indo langsung lahan bisnis


In [86]:
from nltk.tokenize import word_tokenize
tokens = df['text_non_stopword'].tolist()

list_of_list = [word_tokenize(sentence) for sentence in tokens]
list_of_token = [word for sentence in list_of_list for word in sentence]

print(list_of_list)
print(list_of_token)

[['ramadhan', 'idulfitri', 'ada', 'aja', 'program', 'virus', 'ya'], ['penyakit', 'mematikan', 'media'], ['hahahah', 'berita', 'jancok'], ['si', 'media', 'bikin', 'ribet', 'bae', 'cina', 'aja', 'jg', 'ga', 'bikin', 'panik', 'orang', 'klo', 'cari', 'makan', 'yg', 'bener', 'dah', 'heran'], ['teman', 'gua', 'yg', 'tinggal', 'dichina', 'dsna', 'b', 'aja', 'indo', 'aja', 'yg', 'heboh', 'media', 'sosial', 'nya', 'yg', 'sengaja', 'bikin', 'heboh'], ['heran', 'heran', 'pas', 'natal', 'kemana', 'aja', 'giliran', 'ramadhan', 'idul', 'fitri', 'menghampiri', 'ada', 'aja', 'berita', 'sengaja', 'modus', 'rencana', 'orang', 'mengambil', 'keuntungan', 'corona', 'ladang', 'bisnis', 'oknum'], ['media', 'memperkeruh', 'kalo', 'emg', 'anggap', 'aja', 'udah', 'stop', 'hadeh'], ['media', 'emang', 'suka', 'nakutnakutin'], ['yg', 'nama', 'a', 'perubahan', 'cuaca', 'uda', 'sakit', 'imun', 'lemahjustru', 'liat', 'berita', 'kaya', 'gini', 'sakit', 'kefikiran', 'berusaha', 'byak', 'fikiran'], ['indo', 'langsung', 

In [89]:
word_counts = Counter(list_of_token)

for word, count in word_counts.most_common(100):
  print(f"{word}: {count}")

yg: 18
virus: 15
aja: 14
ada: 13
nya: 7
media: 6
berita: 6
bikin: 6
cina: 6
ga: 6
indo: 6
ya: 5
dah: 5
bisnis: 5
china: 5
gue: 5
bilang: 5
orang: 4
heran: 4
lebaran: 4
gak: 4
vaksin: 4
ramadhan: 3
penyakit: 3
makan: 3
udah: 3
sakit: 3
kaya: 3
gini: 3
temen: 3
bodo: 3
percaya: 3
enak: 3
lg: 3
mulu: 3
gw: 3
disana: 3
muw: 3
mematikan: 2
panik: 2
cari: 2
bener: 2
heboh: 2
sengaja: 2
pas: 2
kemana: 2
kalo: 2
cuaca: 2
nanya: 2
bingung: 2
tuh: 2
flu: 2
puasa: 2
beritanya: 2
org: 2
jgn: 2
video: 2
min: 2
hati: 2
maaf: 2
pundi: 2
bawa: 2
meninggal: 2
covid: 2
wabah: 2
rakyat: 2
oprasi: 2
susah: 2
idulfitri: 1
program: 1
hahahah: 1
jancok: 1
si: 1
ribet: 1
bae: 1
jg: 1
klo: 1
teman: 1
gua: 1
tinggal: 1
dichina: 1
dsna: 1
b: 1
sosial: 1
natal: 1
giliran: 1
idul: 1
fitri: 1
menghampiri: 1
modus: 1
rencana: 1
mengambil: 1
keuntungan: 1
corona: 1
ladang: 1
oknum: 1
memperkeruh: 1
emg: 1
anggap: 1
stop: 1


In [94]:
word_dict = {
    "ga":"tidak",
    "gak":"tidak",
    "kalo":"jika",
    "pelakor": "perebut laki orang",
    "bgt" : "sangat",
    "dgn": "dengan",
    "tp":"tapi",
    "sok":"sombong",
    "dr":"dari",
    "gw":"saya",
    'gua':'saya',
    'gue':'saya',
    "bener":"benar",
    "sy":"saya",
    'lo':'kamu',
    'lu':'kamu',
    'org':'orang',
    'n':'dan',
    'pake':'pakai',
    'jd':'jadi',
    'laki':'pria',
    'sm':'sama',
    'yg':'yang',
    'tau':'tahu',
    'd':'di',
    'tolol':'bodoh',
    'bego':'bodoh',
    'att':'atitut',
    'jgn':'jangan',
    'y':'ya',
    'tu':'tuh',
    'krn':'karena',
    'jaman':'zaman',
    'aj':'saja',
    'g':'tidak',
    'now':'sekarang',
    'biar':'agar',
    'geblek':'bodoh',
    'kmrn':'kemarin',
    'skrg':'sekarang',
    'apaoun':'apapun',
    'trus':'lalu',
    "yg":"yang",
    'bodo':'bodoh',
    'jg':'juga'
}

def fix_slang(sentence, word_dict):
  tokens = word_tokenize(sentence)
  fix_slangs = [word_dict.get(word, word) for word in tokens]
  return ' '.join(fix_slangs)


In [95]:
df['text_non_slang']  = df['text_non_stopword'] .apply(lambda x: fix_slang(x, word_dict))
df

,Instagram Comment Text,text_cleaned,text_non_stopword,text_non_slang
1,Setiap mau Ramadhan sama IdulFitri ada aja pro...,setiap mau ramadhan sama idulfitri ada aja pro...,ramadhan idulfitri ada aja program virus ya,ramadhan idulfitri ada aja program virus ya
2,Penyakit mematikan itu adalah Media\n,penyakit mematikan itu adalah media,penyakit mematikan media,penyakit mematikan media
3,Hahahah berita jancok ini\n,hahahah berita jancok ini,hahahah berita jancok,hahahah berita jancok
4,Apaan si media bikin ribet bae di cina biasa2 ...,apaan si media bikin ribet bae di cina biasa a...,si media bikin ribet bae cina aja jg ga bikin ...,si media bikin ribet bae cina aja juga tidak b...
5,Kata teman gua yg tinggal dichina dsna B aja c...,kata teman gua yg tinggal dichina dsna b aja c...,teman gua yg tinggal dichina dsna b aja indo a...,teman saya yang tinggal dichina dsna b aja ind...
6,Heran aku heran tadi pas natal dan tahun baru ...,heran aku heran tadi pas natal dan tahun baru ...,heran heran pas natal kemana aja giliran ramad...,heran heran pas natal kemana aja giliran ramad...
7,Media2 memperkeruh kalo emg di anggap biasa aj...,media memperkeruh kalo emg di anggap biasa aja...,media memperkeruh kalo emg anggap aja udah sto...,media memperkeruh jika emg anggap aja udah sto...
8,Media emang suka nakut-nakutin\n,media emang suka nakutnakutin,media emang suka nakutnakutin,media emang suka nakutnakutin
9,Yg nama a perubahan cuaca uda pasti sakit apa ...,yg nama a perubahan cuaca uda pasti sakit apa ...,yg nama a perubahan cuaca uda sakit imun lemah...,yang nama a perubahan cuaca uda sakit imun lem...
10,di indo langsung jadi lahan bisnis\n,di indo langsung jadi lahan bisnis,indo langsung lahan bisnis,indo langsung lahan bisnis


Bersihlah semua data kita dari Huruf besar kecilnya, tanda baca, stopword dan slang. Baru bisa kita coba cek sentimennya dengan 2 Metode sebelumnya.

## Metode Naive Bayes Komentar Asli

In [98]:
df['count'] = df['text_non_slang'].apply(lambda x: sentiment_counts(x, positive_words, negative_words))
df.head(10)

,Instagram Comment Text,text_cleaned,text_non_stopword,text_non_slang,count
1,Setiap mau Ramadhan sama IdulFitri ada aja pro...,setiap mau ramadhan sama idulfitri ada aja pro...,ramadhan idulfitri ada aja program virus ya,ramadhan idulfitri ada aja program virus ya,-1
2,Penyakit mematikan itu adalah Media\n,penyakit mematikan itu adalah media,penyakit mematikan media,penyakit mematikan media,-1
3,Hahahah berita jancok ini\n,hahahah berita jancok ini,hahahah berita jancok,hahahah berita jancok,0
4,Apaan si media bikin ribet bae di cina biasa2 ...,apaan si media bikin ribet bae di cina biasa a...,si media bikin ribet bae cina aja jg ga bikin ...,si media bikin ribet bae cina aja juga tidak b...,-1
5,Kata teman gua yg tinggal dichina dsna B aja c...,kata teman gua yg tinggal dichina dsna b aja c...,teman gua yg tinggal dichina dsna b aja indo a...,teman saya yang tinggal dichina dsna b aja ind...,0
6,Heran aku heran tadi pas natal dan tahun baru ...,heran aku heran tadi pas natal dan tahun baru ...,heran heran pas natal kemana aja giliran ramad...,heran heran pas natal kemana aja giliran ramad...,-2
7,Media2 memperkeruh kalo emg di anggap biasa aj...,media memperkeruh kalo emg di anggap biasa aja...,media memperkeruh kalo emg anggap aja udah sto...,media memperkeruh jika emg anggap aja udah sto...,0
8,Media emang suka nakut-nakutin\n,media emang suka nakutnakutin,media emang suka nakutnakutin,media emang suka nakutnakutin,0
9,Yg nama a perubahan cuaca uda pasti sakit apa ...,yg nama a perubahan cuaca uda pasti sakit apa ...,yg nama a perubahan cuaca uda sakit imun lemah...,yang nama a perubahan cuaca uda sakit imun lem...,-1
10,di indo langsung jadi lahan bisnis\n,di indo langsung jadi lahan bisnis,indo langsung lahan bisnis,indo langsung lahan bisnis,0


In [99]:
df['hasil sentimen'] = df['count'].apply(sentimen)
df

,Instagram Comment Text,text_cleaned,text_non_stopword,text_non_slang,count,hasil sentimen
1,Setiap mau Ramadhan sama IdulFitri ada aja pro...,setiap mau ramadhan sama idulfitri ada aja pro...,ramadhan idulfitri ada aja program virus ya,ramadhan idulfitri ada aja program virus ya,-1,negatif
2,Penyakit mematikan itu adalah Media\n,penyakit mematikan itu adalah media,penyakit mematikan media,penyakit mematikan media,-1,negatif
3,Hahahah berita jancok ini\n,hahahah berita jancok ini,hahahah berita jancok,hahahah berita jancok,0,netral
4,Apaan si media bikin ribet bae di cina biasa2 ...,apaan si media bikin ribet bae di cina biasa a...,si media bikin ribet bae cina aja jg ga bikin ...,si media bikin ribet bae cina aja juga tidak b...,-1,negatif
5,Kata teman gua yg tinggal dichina dsna B aja c...,kata teman gua yg tinggal dichina dsna b aja c...,teman gua yg tinggal dichina dsna b aja indo a...,teman saya yang tinggal dichina dsna b aja ind...,0,netral
6,Heran aku heran tadi pas natal dan tahun baru ...,heran aku heran tadi pas natal dan tahun baru ...,heran heran pas natal kemana aja giliran ramad...,heran heran pas natal kemana aja giliran ramad...,-2,negatif
7,Media2 memperkeruh kalo emg di anggap biasa aj...,media memperkeruh kalo emg di anggap biasa aja...,media memperkeruh kalo emg anggap aja udah sto...,media memperkeruh jika emg anggap aja udah sto...,0,netral
8,Media emang suka nakut-nakutin\n,media emang suka nakutnakutin,media emang suka nakutnakutin,media emang suka nakutnakutin,0,netral
9,Yg nama a perubahan cuaca uda pasti sakit apa ...,yg nama a perubahan cuaca uda pasti sakit apa ...,yg nama a perubahan cuaca uda sakit imun lemah...,yang nama a perubahan cuaca uda sakit imun lem...,-1,negatif
10,di indo langsung jadi lahan bisnis\n,di indo langsung jadi lahan bisnis,indo langsung lahan bisnis,indo langsung lahan bisnis,0,netral


In [100]:
df['hasil sentimen'].value_counts()

,count
hasil sentimen,
netral,27
negatif,20
positif,3


Dengan Naive Bayes, Sentimen terbanyak di data kita adalah sentimen Netral, yang diikuti sentimen Negatif dan terakhir di Positif.

## Metode Lexicon Komentar Asli

In [101]:
df['lexicon_sentiment'] = df['text_non_slang'].apply(lambda x: sentiment_counts_lex(x, positive_lex, negative_lex, stemmer))
df

,Instagram Comment Text,text_cleaned,text_non_stopword,text_non_slang,count,hasil sentimen,lexicon_sentiment
1,Setiap mau Ramadhan sama IdulFitri ada aja pro...,setiap mau ramadhan sama idulfitri ada aja pro...,ramadhan idulfitri ada aja program virus ya,ramadhan idulfitri ada aja program virus ya,-1,negatif,negative
2,Penyakit mematikan itu adalah Media\n,penyakit mematikan itu adalah media,penyakit mematikan media,penyakit mematikan media,-1,negatif,negative
3,Hahahah berita jancok ini\n,hahahah berita jancok ini,hahahah berita jancok,hahahah berita jancok,0,netral,neutral
4,Apaan si media bikin ribet bae di cina biasa2 ...,apaan si media bikin ribet bae di cina biasa a...,si media bikin ribet bae cina aja jg ga bikin ...,si media bikin ribet bae cina aja juga tidak b...,-1,negatif,negative
5,Kata teman gua yg tinggal dichina dsna B aja c...,kata teman gua yg tinggal dichina dsna b aja c...,teman gua yg tinggal dichina dsna b aja indo a...,teman saya yang tinggal dichina dsna b aja ind...,0,netral,positive
6,Heran aku heran tadi pas natal dan tahun baru ...,heran aku heran tadi pas natal dan tahun baru ...,heran heran pas natal kemana aja giliran ramad...,heran heran pas natal kemana aja giliran ramad...,-2,negatif,positive
7,Media2 memperkeruh kalo emg di anggap biasa aj...,media memperkeruh kalo emg di anggap biasa aja...,media memperkeruh kalo emg anggap aja udah sto...,media memperkeruh jika emg anggap aja udah sto...,0,netral,neutral
8,Media emang suka nakut-nakutin\n,media emang suka nakutnakutin,media emang suka nakutnakutin,media emang suka nakutnakutin,0,netral,neutral
9,Yg nama a perubahan cuaca uda pasti sakit apa ...,yg nama a perubahan cuaca uda pasti sakit apa ...,yg nama a perubahan cuaca uda sakit imun lemah...,yang nama a perubahan cuaca uda sakit imun lem...,-1,negatif,negative
10,di indo langsung jadi lahan bisnis\n,di indo langsung jadi lahan bisnis,indo langsung lahan bisnis,indo langsung lahan bisnis,0,netral,neutral


In [102]:
df['lexicon_sentiment'].value_counts()

,count
lexicon_sentiment,
negative,22
neutral,19
positive,9


Ternyata berbeda dengan hasil sebelumnya, Lexicon memberikan pernyataan bahwa Sentimen Negatif lebih banyak daripada Netral dan tetap Positif ada di peringkat terakhir.

## Hasil Penelitian Penggunaan Metode Naive Bayes dan Lexicon Pada Komentar IG Indozone ID



Komentar Pada IG Indozone ID pada dasarnya tergantung berita pada postingannya. dan jika dilihat langsung dari komentarnya, Metode Lexicon dapat menilai dengan lebih baik dibandingkan Naive Bayes jika berhubungan dengan sentimen yang terbatas.


In [103]:
kolom_terpilih = ['Instagram Comment Text', 'hasil sentimen', 'lexicon_sentiment']
df_ = df[kolom_terpilih].copy()
df_

,Instagram Comment Text,hasil sentimen,lexicon_sentiment
1,Setiap mau Ramadhan sama IdulFitri ada aja pro...,negatif,negative
2,Penyakit mematikan itu adalah Media\n,negatif,negative
3,Hahahah berita jancok ini\n,netral,neutral
4,Apaan si media bikin ribet bae di cina biasa2 ...,negatif,negative
5,Kata teman gua yg tinggal dichina dsna B aja c...,netral,positive
6,Heran aku heran tadi pas natal dan tahun baru ...,negatif,positive
7,Media2 memperkeruh kalo emg di anggap biasa aj...,netral,neutral
8,Media emang suka nakut-nakutin\n,netral,neutral
9,Yg nama a perubahan cuaca uda pasti sakit apa ...,negatif,negative
10,di indo langsung jadi lahan bisnis\n,netral,neutral


Adakah Cyber Bullying yang terjadi di Akun IG Indozone ID? jelas sekali ada karena masih banyak komentar negatif yang ada di kedua metode walaupun dengan Naive Bayes ada di urutan kedua, sedangkan dengan Lexicon ada di urutan pertama

In [105]:
df_['hasil sentimen'].value_counts()


,count
hasil sentimen,
netral,27
negatif,20
positif,3


In [104]:
df_['lexicon_sentiment'].value_counts()


,count
lexicon_sentiment,
negative,22
neutral,19
positive,9


Dari 2 preview diatas bisa menunjukkan bahwa metode yang berbeda walaupun dengan dataset yang sama bisa memiliki hasil yang berbeda juga

##Kesimpulan

Kesimpulan yang dapat diambil dari penelitian diatas adalah banyaknya data latihan dapat menentukan perbedaan besar hasil dari Sentiment Analysis.

Dengan 400 training data yang saya dapatkan dari hasil googling, kedua Metode pasti Memiliki hasil yang berbeda dan itu tetap tergantung seberapa banyak data training yang dapat kita gunakan untuk melatih metode tersebut.

Dan di Penelitian ini bisa saya katakan bahwa cyberBullying akan tetap ada (walaupun bukan mayoritas tapi tetap bisa memperburuk keadaan) jika kita tidak menanamkan sifat positif pada diri kita. Sebarkan positifitas kita ke orang-orang di sekitar dan terus lakukan itu hingga semua orang dapat lebih memikirkan hal Positif daripada negatif.